src/node.cc

```diff
@@ -1291,12 +1236,6 @@ void SetupPromises(const FunctionCallbackInfo<Value>& args) {
 }  // anonymous namespace
 
 
-void AddPromiseHook(v8::Isolate* isolate, promise_hook_func fn, void* arg) {
-  Environment* env = Environment::GetCurrent(isolate);
-  env->AddPromiseHook(fn, arg);
-}
```

Comment:
If this lands in Node 8, then at least our need for `AddPromiseHook` is gone. It is only if these promise hooks don't make it into async_wrap in Node 8 as publicly documented features that we needed to share the promise hook.